### PR TITLE
OCPQE-24474  add z-stream release detector

### PR DESCRIPTION
**Pull request by rioliu-rh**
```console
$ oarctl start-release-detector -h
Usage: oarctl start-release-detector [OPTIONS]

Options:
  -r, --minor-release TEXT  Minor rlease of OCP e.g. 4.y  [required]
  -h, --help                Show this message and exit.
```
check new z-stream release for OCP minor release
```console
$ oarctl start-release-detector -r 4.15
2024-08-06T15:24:54Z: INFO: latest z-stream version: 4.15.25
2024-08-06T15:24:54Z: INFO: latest stable version: 4.15.25
2024-08-06T15:24:54Z: INFO: no new z-stream release found
```

### CODE REVIEW

File: oar/controller/detector.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+from oar.cli.cmd_create_test_report import create_test_report
+
+logger = logging.getLogger(__name__)
+
+
+class ReleaseDetector:
+
+    def __init__(self, minor_release):
+        """
+        Comopare latest z-stream version from file generated by ART and latest stable version
```
Comment (tomasdavidorg) [inline oar/controller/detector.py:16]:
```suggestion
        Compare the latest z-stream version from file generated by ART and the latest stable version
```

File: oar/controller/detector.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        from release stream, if they're different, it mean new z-stream release resources are
```
Comment (tomasdavidorg) [inline oar/controller/detector.py:17]:
```suggestion
        from release stream, if they're different, it means the new z-stream release resources are
```

File: oar/controller/detector.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        prepared, we can kick off QE release flow
+        """
+        self._minor_release = minor_release
+        # TODO: create statebox object
+
+    def get_latest_zstream_version(self):
+        """
+        Get latest z-stream version from file generated by ART
```
Comment (tomasdavidorg) [inline oar/controller/detector.py:26]:
```suggestion
        Get the latest z-stream version from file generated by ART
```

File: oar/controller/detector.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        resp = requests.get(url)
+        if resp.ok:
+            yamlobj = yaml.safe_load(resp.text)
+            return list(yamlobj["releases"].keys())[-1]
+        else:
+            return None
+
+    def get_latest_stable_version(self):
+        """
+        Get latest stable version from release stream
```
Comment (tomasdavidorg) [inline oar/controller/detector.py:38]:
```suggestion
        Get the latest stable version from release stream
```

**Comment by rioliu-rh** [comment]
add func to validate input version
```console
$ oarctl start-release-detector -r 5.1
Usage: oarctl start-release-detector [OPTIONS]
Try 'oarctl start-release-detector -h' for help.

Error: Invalid value for '-r' / '--minor-release': Invalid OCP minor version 5.1
```

**Comment by rioliu-rh** [comment]
@ming1013 Can you review again, thx

**Review by tomasdavidorg — COMMENTED** [review]
Just some minor typos.

**Comment by rioliu-rh** [comment]
> Just some minor typos.

refined the comments

**Pull request merged**